shared-bindings/socketpool/Socket.c

```diff
@@ -255,6 +255,39 @@ STATIC mp_obj_t _socketpool_socket_send(mp_obj_t self_in, mp_obj_t buf_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(socketpool_socket_send_obj, _socketpool_socket_send);
 
+//|     def sendall(self, bytes: ReadableBuffer) -> None:
+//|         """Send some bytes to the connected remote address.
+//|         Suits sockets of type SOCK_STREAM
+//|
+//|         This calls send() repeatedly until all the data is sent or an error
+//|         occurs. If an error occurs, it's impossible to tell how much data
+//|         has been sent.
+//|
+//|         :param ~bytes bytes: some bytes to send"""
+//|         ...
+STATIC mp_obj_t _socketpool_socket_sendall(mp_obj_t self_in, mp_obj_t buf_in) {
+    socketpool_socket_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    if (common_hal_socketpool_socket_get_closed(self)) {
+        // Bad file number.
+        mp_raise_OSError(MP_EBADF);
+    }
+    if (!common_hal_socketpool_socket_get_connected(self)) {
+        mp_raise_BrokenPipeError();
+    }
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(buf_in, &bufinfo, MP_BUFFER_READ);
+    while (bufinfo.len > 0) {
+        mp_int_t ret = common_hal_socketpool_socket_send(self, bufinfo.buf, bufinfo.len);
```

Comment:
Might this block and block background tasks? Should there be a `RUN_BACKGROUND_TASKS` here?

Reply:
Added